src/components/FileTreeNode/styles.module.scss

```diff
@@ -2,9 +2,32 @@
 
 .node {
   cursor: pointer;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: nowrap;
+  justify-content: space-between;
   padding: 0.5rem 1.25rem;
 }
 
+.nodeName {
+  word-break: break-all;
```

Comment:
These styles don't look so nice on laptop-sized windows. How about this patch? It's a bit of an improvement.

<details>
<summary>patch</summary>

```diff
diff --git a/src/components/FileTreeNode/index.tsx b/src/components/FileTreeNode/index.tsx
index d02a372..7067676 100644
--- a/src/components/FileTreeNode/index.tsx
+++ b/src/components/FileTreeNode/index.tsx
@@ -134,23 +134,17 @@ class FileTreeNodeBase<TreeNodeType> extends React.Component<PublicProps> {
       <React.Fragment>
         <ListGroup.Item {...listGroupItemProps}>
           <span
-            className={styles.nodeName}
-            style={{ paddingLeft: `${level * 20}px` }}
+            className={styles.nodeItem}
+            style={{ paddingLeft: `${level * 12}px` }}
           >
             {isFolder ? (
-              <React.Fragment>
-                <FontAwesomeIcon icon={isExpanded ? 'folder-open' : 'folder'} />
-                &nbsp;{node.name}
-              </React.Fragment>
+              <FontAwesomeIcon icon={isExpanded ? 'folder-open' : 'folder'} />
             ) : (
-              <React.Fragment>
-                <FontAwesomeIcon icon="file" />
-                &nbsp;{node.name}
-              </React.Fragment>
+              <FontAwesomeIcon icon="file" />
             )}
+            <span className={styles.nodeName}>{node.name}</span>
+            {nodeIcons}
           </span>
-
-          {nodeIcons}
         </ListGroup.Item>
 
         {isExpanded &&
diff --git a/src/components/FileTreeNode/styles.module.scss b/src/components/FileTreeNode/styles.module.scss
index 54d97e8..e0dd03b 100644
--- a/src/components/FileTreeNode/styles.module.scss
+++ b/src/components/FileTreeNode/styles.module.scss
@@ -9,8 +9,29 @@
   padding: 0.5rem 1.25rem;
 }
 
+.nodeItem {
+  align-items: flex-start;
+  display: flex;
+  line-height: 1;
+  width: 100%;
+}
+
+.nodeName,
+.nodeIcons {
+  margin-left: 0.5rem;
+  width: 100%;
+}
+
 .nodeName {
-  word-break: break-all;
+  overflow: hidden;
+  padding-bottom: 0.2rem;
+  text-overflow: ellipsis;
+  white-space: wrap;
+}
+
+.nodeIcons {
+  display: flex;
+  justify-content: flex-end;
 }
 
 .selected {
@@ -28,6 +49,3 @@
     color: $orange;
   }
 }
-
-.nodeIcons {
-}
```

</details>

<img width="568" alt="Screenshot 2019-03-20 12 57 59" src="https://user-images.githubusercontent.com/55398/54707850-0e1f4100-4b10-11e9-8852-7989d462458c.png">
<img width="965" alt="Screenshot 2019-03-20 12 57 44" src="https://user-images.githubusercontent.com/55398/54707852-0f506e00-4b10-11e9-9b61-7adc10a0a22d.png">


I filed https://github.com/mozilla/addons-code-manager/issues/431 to make a storybook out of this.

Reply:
oh yes, I love it! Thanks ❤️